File: src/admin_app/lib/core/models/branch.dart
import 'localized_text.dart';

class Branch {
  final int id;
  final LocalizedText name;
  final LocalizedText? address;
  final String? phone;
  final bool isActive;
  final int displayOrder;

  const Branch({
    required this.id,
    required this.name,
    this.address,
    this.phone,
    required this.isActive,
    required this.displayOrder,
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['id'] as int,
      name: LocalizedText.parse(json['name']),
      address: LocalizedText.parseNullable(json['address']),
      phone: json['phone'] as String?,
      isActive: json['isActive'] as bool? ?? true,
      displayOrder: json['displayOrder'] as int? ?? 0,
    );
  }
}

File: src/Branch.API/Model/Branch.cs
namespace Chillax.Branch.API.Model;

public class Branch
{
    public int Id { get; set; }
    public LocalizedText Name { get; set; } = new();
    public LocalizedText? Address { get; set; }
    public string? Phone { get; set; }
    public bool IsActive { get; set; } = true;
    public int DisplayOrder { get; set; }
    public TimeOnly DayStartTime { get; set; } = new(17, 0);
    public TimeOnly DayEndTime { get; set; } = new(5, 0);
    public bool IsOrderingEnabled { get; set; } = true;
    public bool IsReservationsEnabled { get; set; } = true;
}
